src/android/com/microsoftopentechnologies/intellij/wizards/createvm/CloudServiceStep.java

```diff
@@ -0,0 +1,322 @@
+ *  limitations under the License.
+ */
+
+package com.microsoftopentechnologies.intellij.wizards.createvm;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.ui.components.JBScrollPane;
+import com.intellij.ui.wizard.WizardNavigationState;
+import com.intellij.ui.wizard.WizardStep;
+import com.microsoftopentechnologies.intellij.helpers.UIHelper;
+import com.microsoftopentechnologies.intellij.helpers.azure.AzureCmdException;
+import com.microsoftopentechnologies.intellij.helpers.azure.sdk.AzureSDKManagerImpl;
+import com.microsoftopentechnologies.intellij.model.vm.CloudService;
+import com.microsoftopentechnologies.intellij.model.vm.StorageAccount;
+import com.microsoftopentechnologies.intellij.model.vm.VirtualMachineImage;
+
+import javax.swing.*;
+import javax.swing.event.HyperlinkEvent;
+import javax.swing.event.HyperlinkListener;
+import java.awt.*;
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.text.DateFormat;
+import java.util.List;
+import java.util.Vector;
+
+public class CloudServiceStep extends WizardStep<CreateVMWizardModel> {
+
+    private Project project;
+    private CreateVMWizardModel model;
+    private JPanel rootPanel;
+    private JList createVmStepsList;
+    private JEditorPane imageDescriptionTextPane;
+    private JComboBox cloudServiceComboBox;
+    private JComboBox storageComboBox;
+    private JCheckBox availabilitySetCheckBox;
+    private JComboBox availabilityComboBox;
+    private List<StorageAccount> storageAccounts;
+    private List<CloudService> cloudServices;
+
+
+    public CloudServiceStep(CreateVMWizardModel mModel, final Project project) {
+
+        super("Cloud Service Settings");
+
+        this.project = project;
+        this.model = mModel;
+
+        model.configStepList(createVmStepsList, 3);
+
+        cloudServiceComboBox.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent itemEvent) {
+                fillStorage();
+                fillAvailabilitySets();
+            }
+        });
+
+        cloudServiceComboBox.setRenderer(new DefaultListCellRenderer() {
+            @Override
+            public Component getListCellRendererComponent(JList<?> jList, Object o, int i, boolean b, boolean b1) {
+                if (o instanceof CloudService) {
+                    CloudService sa = (CloudService) o;
+                    return super.getListCellRendererComponent(jList, String.format("%s (%s)", sa.getName(), sa.getLocation()), i, b, b1);
+                } else {
+                    return super.getListCellRendererComponent(jList, o, i, b, b1);
+                }
+            }
+        });
+
+        storageComboBox.setRenderer(new DefaultListCellRenderer() {
```

Comment:
OK. According to the inspection this causes issues only in GTK+. Maybe it'll be only on GNOME based desktops on Linux?